DataCollector/DoctrineDataCollector.php

```diff
@@ -101,7 +101,7 @@ public function collect(Request $request, Response $response, Throwable $excepti
             $cacheConfiguration = $emConfig->getSecondLevelCacheConfiguration();
             assert($cacheConfiguration instanceof CacheConfiguration);
             $cacheLoggerChain = $cacheConfiguration->getCacheLogger();
-            assert($cacheLoggerChain instanceof CacheLoggerChain);
+            assert($cacheLoggerChain instanceof CacheLoggerChain || $cacheConfiguration === null);
```

Comment:
Instead of the assert, the `if (! $cacheLoggerChain)` below could be changed to `! $cacheLoggerChain instanceof CacheLoggerChain)` (which would also cover a case where a custom CacheLogger is passed without using a chain)

Reply:
I don't like that because it would silence the error in case of unexpected instance.

Reply:
The thing is, the data collector is not in charge of configuring the cache logger. If the debugging tool breaks entirely because you customized the config in a way that they did not expect, that's quite bad. It seems better to skip the SLC stats in such case.

Reply:
I'm more proponent of [offensive programming](https://en.wikipedia.org/wiki/Offensive_programming) and there it's better to crash than display incorrect information.